ftfy/fixes.py

```diff
@@ -192,22 +194,21 @@ def fix_one_step_and_explain(text):
                 # Check encoded_bytes for sequences that would be UTF-8,
                 # except they have b' ' where b'\xa0' would belong.
                 if ALTERED_UTF8_RE.search(encoded_bytes):
-                    fixed_bytes, cost = restore_byte_a0(encoded_bytes)
-                    try:
-                        fixed = fixed_bytes.decode('utf-8-variants')
-                        steps = [('encode', encoding,
-                                  ENCODING_COSTS.get(encoding, 0)),
-                                 ('transcode', 'restore_byte_a0', cost),
-                                 ('decode', 'utf-8-variants', 0)]
-                        return fixed, steps
-                    except UnicodeDecodeError:
-                        pass
+                    encoded_bytes, cost = _restore_byte_a0_with_cost(encoded_bytes)
+                    transcode_steps.append(('transcode', 'restore_byte_a0', cost))
+
+                # Check for the byte 0x1a, which indicates where one of our
+                # 'sloppy' codecs found a replacement character.
+                if b'\x1a' in encoded_bytes:
```

Comment:
I've just noticed: I should only do this in cases where we actually used a sloppy codec. If we didn't, it means the byte 1A was there from the start (for whatever reason) and I should leave it alone.